test/parallel/test-util-format.js

```diff
@@ -269,6 +269,26 @@ assert.strictEqual(util.format('%s', -Infinity), '-Infinity');
   );
 }
 
+// Symbol.toPrimitive handling for string format specifier
+{
+  const objectWithToPrimitive = {
+    [Symbol.toPrimitive](hint) {
+      switch (hint) {
+        case 'number':
+          return 42;
+        case 'string':
+          return 'string representation';
+        case 'default':
+        default:
+          return 'default context';
+      }
+    }
+  };
+
+  // TODO Add tests here after fixing the same bug for %d/%j.
```

Comment:
I am not certain this comment is correct? %j is custom and it is meant to use `JSON.stringify()` which it does and %d is only [for numbers and works as specified](https://console.spec.whatwg.org/#formatter). I would just remove it?

```suggestion
```

Reply:
Thank you, I think you are right. I've removed the comments here and added test cases for 'number' and 'default'.